### ISSUE TITLE
Filtering of tests with use of C library starknet_crypto

### ISSUE THREAD

**Issue by edisontim** [keep-starknet-strange/ziggy-starkdust#254]
When using ```zig build tests```
with these lines added to the build.zig file: 
```
    unit_tests.addIncludePath(.{ .path = "./src/math/crypto/starknet_crypto/" });
    unit_tests.addObjectFile(std.build.LazyPath{ .path = "./src/math/crypto/starknet_crypto/libstarknet_crypto.a" });
```
(these lines are necessary for the linker to include the C starknet_crypto library for the pedersen_hash and to include the header) my tests are running smoothly but when I run ```zig test src/tests.zig``` I get the following error:
```src/math/crypto/pedersen_hash.zig:5:25: error: C import failed
const starknet_crypto = @cImport(@cInclude("starknet_crypto.h"));
                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
referenced by:
    pedersenHash: src/math/crypto/pedersen_hash.zig:13:5
    pedersen_hash: src/vm/builtins/builtin_runner/hash.zig:19:72
    remaining reference traces hidden; use '-freference-trace' to see all reference traces
/Users/timothyedison/Documents/Code/cairo-zig-tedison/zig-cache/o/f1574732c51f72a3eb83fdbd90903638/cimport.h:1:10: error: 'starknet_crypto.h' file not found
#include <starknet_crypto.h>
```
I'm guessing it's because the zig test command does not use the build.zig file. 
Options are: 
1. using a script that hides the command ```zig test -I ./src/math/crypto/starknet_crypto/ -L./src/math/crypto/starknet_crypto -lstarknet_crypto src/tests.zig --test-filter "$@"``` which adds include path and library path to the tests
2. find a way to filter the tests when using ```zig build tests```

**Comment by jobez** [comment]
Looking into tigerbeetle, we should be able to add a param to the `build.zig` as so https://github.com/tigerbeetle/tigerbeetle/blob/main/build.zig#L208 which then gets handled to their test step here:https://github.com/tigerbeetle/tigerbeetle/blob/main/build.zig#L231

We would want the same flow for our `b.addTest` https://github.com/keep-starknet-strange/cairo-zig/blob/main/build.zig#L123

We would also want to update the readme here to choose this route for filtering: https://github.com/keep-starknet-strange/cairo-zig/blob/main/README.md?plain=1#L88

Would you want to add this?

**Comment by edisontim** [comment]
Yes I can take this :)

**Comment by edisontim** [comment]
Closed by #244
